docs/running-on-kubernetes.md

```diff
@@ -561,6 +561,13 @@ specific to Spark on Kubernetes.
     This is distinct from <code>spark.executor.cores</code>: it is only used and takes precedence over <code>spark.executor.cores</code> for specifying the executor pod cpu request if set. Task 
     parallelism, e.g., number of tasks an executor can run concurrently is not affected by this.
 </tr>
+<tr>
+  <td><code>spark.kubernetes.executor.maxInitFailures</code></td>
+  <td>10</td>
+  <td>
+    Maximum number of times executors are allowed to fail with an Init:Error state before failing the application. Note that Init:Error failures should not be caused by Spark itself because Spark does not attach init-containers to pods. Init-containers can be attached by the cluster itself. Users should check with their cluster administrator if these kinds of failures to start the executor pod occur frequently.
```

Comment:
This is very specific and covering exactly one kind of error. I'd like this property to cover all initialization errors.

Reply:
Also, I would change the description here, because using init containers for injecting init-containers through mutable webhooks is not something that's all that common. Also should be linked to https://kubernetes.io/docs/admin/extensible-admission-controllers/

Reply:
I'm in agreement w/ @foxish about designing for wider categories of error, which can be future-proofed

Reply:
As per https://github.com/apache/spark/pull/21241#discussion_r186789848 I think it's important to define what that full set of error types is. Do we have a comprehensive list we can follow?

Reply:
We can also just start with a minimal set and just keep adding them as we find more root causes.